src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/FixLegacyResourceDesignerStep.cs

```diff
@@ -0,0 +1,158 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+
+using Java.Interop.Tools.Cecil;
+
+using Mono.Linker;
+using Mono.Linker.Steps;
+
+using Mono.Tuner;
+#if ILLINK
+using Microsoft.Android.Sdk.ILLink;
+#endif  // ILLINK
+
+namespace MonoDroid.Tuner
+{
+	public class FixLegacyResourceDesignerStep : LinkDesignerBase
+	{
+		const string DesignerAssemblyName = "_Microsoft.Android.Resource.Designer";
+#if ILLINK
+		protected override void Process ()
+		{
+			cache = Context;
+		}
+#else
```

Comment:
Convention: `#else` and `#endif` should have comments mentioning the source expression, e.g.

```csharp
#if ILLINK
…
#else   // !ILLINK
#endif  // !ILLINK
```